examples/pipeline_tacotron2/inference.py

```diff
@@ -0,0 +1,294 @@
+        waveglow = waveglow.remove_weightnorm(waveglow)
+        waveglow = waveglow.to(device)
+        waveglow.eval()
+
+        if args.jit:
+            raise ValueError("Vocoder option `nvidia_waveglow is not jittable.")
+
+        with torch.no_grad():
+            waveform = waveglow.infer(mel_specgram).cpu()
+
+    elif args.vocoder == "wavernn":
+        from torchaudio.models import wavernn
+        sys.path.append(os.path.join(os.path.dirname(__file__), "../pipeline_wavernn"))
+        from wavernn_inference_wrapper import WaveRNNInferenceWrapper
+        from processing import NormalizeDB
+
+        wavernn_model = wavernn(args.wavernn_checkpoint_name).eval().to(device)
+        wavernn_inference_model = WaveRNNInferenceWrapper(wavernn_model)
+
+        if args.jit:
+            wavernn_inference_model = torch.jit.script(wavernn_inference_model)
+
+        # WaveRNN spectro setting for default checkpoint
+        # n_fft = 2048
+        # n_mels = 80
+        # win_length = 1100
+        # hop_length = 275
+        # f_min = 40
+        # f_max = 11025
+
+        transforms = torch.nn.Sequential(
+            InverseSpectralNormalization(),
+            NormalizeDB(min_level_db=-100, normalization=True),
+        )
+        mel_specgram = transforms(mel_specgram.cpu())
+
+        with torch.no_grad():
+            waveform = wavernn_inference_model(mel_specgram.to(device),
+                                               loss_name=args.wavernn_loss,
+                                               mulaw=(not args.wavernn_no_mulaw),
+                                               batched=(not args.wavernn_no_batch_inference),
+                                               timesteps=args.wavernn_batch_timesteps,
+                                               overlap=args.wavernn_batch_overlap,)
+        waveform = waveform.unsqueeze(0)
+
+    elif args.vocoder == "griffin_lim":
+        from torchaudio.transforms import GriffinLim, InverseMelScale
+
+        inv_norm = InverseSpectralNormalization()
+        inv_mel = InverseMelScale(
+            n_stft=(args.n_fft // 2 + 1),
+            n_mels=args.n_mels,
+            sample_rate=args.sample_rate,
+            f_min=args.mel_fmin,
+            f_max=args.mel_fmax,
+            mel_scale="slaney",
+            norm='slaney',
+        )
+        griffin_lim = GriffinLim(
+            n_fft=args.n_fft,
+            power=1,
+            hop_length=256,
+            win_length=1024,
+        )
+
+        vocoder = torch.nn.Sequential(
+            inv_norm,
+            inv_mel,
+            griffin_lim
+        )
+
+        if args.jit:
+            vocoder = torch.jit.script(vocoder)
+
+        waveform = vocoder(mel_specgram.cpu())
+
+    torchaudio.save(args.output_path, waveform, args.sample_rate)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='TTS Generator')
```

Comment:
What I often do when writing a script is;
1. Add a module-level docstring.
2. Refer the docstring `__doc__` from `ArgumentParser(description=__doc__)`.
That way, one can write detailed docstring which is accessible from CLI as well.

(Also I would put the construction of parer in `parse_args` as well.)

Reply:
Thanks for the reminder, I've move the parser into `parse_args` and wrote a simple module docstring.